src/layout/Popup.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "../layout/Surface", "src/form/Input", "../c3chart/Column", "css!./Popup", "css!font-awesome",], factory);
+    } else {
+        root.layout_Popup = factory(root.d3, root.common_HTMLWidget, root.layout_Surface, root.form_Input, root.c3chart_Column);
+    }
+}(this, function (d3, HTMLWidget, Surface, Input, Column) {
+    function Popup() {
+        HTMLWidget.call(this);
+
+        this._tag = "div";
+
+        this._surfaceButtons = [];
+    }
+    Popup.prototype = Object.create(HTMLWidget.prototype);
+    Popup.prototype._class += " layout_Popup";
+    
+    Popup.prototype.publish("showTestButton", false, "boolean", "Test Button in target",null,{});
+    Popup.prototype.publish("popupState", false, "boolean", "State of the popup, visible (true) or hidden (false)",null,{});
+    Popup.prototype.publish("width", null, "number", "Popup width",null,{});
```

Comment:
width and height are already used in the base class Widget - overriding them will probably cause pain.